include/event_manager.rst

```diff
@@ -186,15 +137,74 @@ The following code example shows a source file for the event type ``sample_event
 		  	  log_sample_event, 	/* Function logging event data. */
 		  	  NULL); 		/* No event info provided. */
 
+Submitting an event
+===================
+
+To submit an event of a given type (for example, ``sample_event``), you must first allocate it by calling the function with the name new\_\ *event_type_name* (for example, ``new_sample_event()``).
+You can then write values to the data fields.
+Finally, use :c:macro:`EVENT_SUBMIT` to submit the event.
+
+The following code example shows how to create and submit an event of type ``sample_event`` that has three data fields:
+
+.. code-block:: c
+
+	/* Allocate event. */
+	struct sample_event *event = new_sample_event();
+
+	/* Write data to datafields. */
+	event->value1 = value1;
+	event->value2 = value2;
+	event->value3 = value3;
+
+	/* Submit event. */
+	EVENT_SUBMIT(event);
+
+If an event type also defines data with variable size, you must pass also the size of the data as an argument to the function that allocates the event.
+For example, if the ``sample_event`` also contains data with variable size, you must apply the following changes to the code:
+
+.. code-block:: c
 
+	/* Allocate event. */
+	struct sample_event *event = new_sample_event(my_data_size);
+
+	/* Write data to datafields. */
+	event->value1 = value1;
+	event->value2 = value2;
+	event->value3 = value3;
+
+	/* Write data with variable size. */
+	memcpy(event->dyndata.data, my_buf, my_data_size);
+
+	/* Submit event. */
+	EVENT_SUBMIT(event);
+
+After the event is submitted, the Event Manager adds it to the processing queue.
+When the event is processed, the Event Manager notifies all modules that subscribe to this event type.
+
+.. warning::
+
+	Events are dynamically allocated and must be submitted.
+	If an event is not submitted, it will not be handled and the memory will not be freed.
+
+Modules
+*******
+
+Modules are separated source file that can subscribe to every defined event.
+Each module can subscribe to multiple events, there is no any limitation.
+An application can have as many modules as required.
+
+Event manager provide API for subscribing modules to a specific events defined in the application.
+If a module is subscribing to a specific event, its called listener.
+Every listener is identified by a unique name.
```

Comment:
I think these two lines should be in line 52, and the line from the line 52 should go here.

Reply:
Won't do. Discussed offline.